lib/srv/forward/sshserver.go

```diff
@@ -862,28 +940,56 @@ func (s *Server) rejectChannel(chans <-chan ssh.NewChannel, errMessage string) {
 }
 
 func (s *Server) handleGlobalRequest(ctx context.Context, req *ssh.Request) {
-	// Version requests are internal Teleport requests, they should not be
-	// forwarded to the remote server.
-	if req.Type == teleport.VersionRequest {
+	switch req.Type {
+	case teleport.VersionRequest:
+		// Version requests are internal Teleport requests, they should not be
+		// forwarded to the remote server.
 		err := req.Reply(true, []byte(teleport.Version))
 		if err != nil {
 			s.log.Debugf("Failed to reply to version request: %v.", err)
 		}
 		return
+	case teleport.TCPIPForwardRequest, teleport.CancelTCPIPForwardRequest:
+		// Forwarding requests need to be authorized first.
+		if err := s.checkTCPIPForwardRequest(req); err != nil {
+			s.log.WithError(err).Warnf("Failed to check tcpip forward request")
+			if err := req.Reply(false, nil); err != nil {
+				s.log.Warnf("Failed to reply to global request: %v: %v", req.Type, err)
+			}
+			return
+		}
+		// Pass request on unchanged.
+	case teleport.KeepAliveReqType:
+	default:
+		s.log.Debugf("Discarding %q global request: %+v", req.Type, req)
+		return
```

Comment:
The SSH state machine doesn't allow for discarding global requests that want a reply:

(I'm not super convinced that `%+v` looks clean enough for logging on a `ssh.Request` but I don't really know if it's worth logging the payload for the request anyway)

```suggestion
		s.log.Debugf("Rejecting unknown global request %q.", req.Type)
		_ = req.Reply(false, nil)
		return
```